packages/api/src/utils/stripe.js

```diff
@@ -0,0 +1,124 @@
+import Stripe from 'stripe'
+
+/**
+ * @typedef {import('stripe').Stripe} StripeInterface
+ */
+
+/**
+ * @typedef {import("./billing-types").BillingService} BillingService
+ */
+
+/**
+ * A BillingService that uses stripe.com
+ */
+export class StripeBillingService {
+  static create () {
+    return new StripeBillingService()
+  }
+
+  /**
+   * @protected
+   */
+  constructor () {
+    /**
+     * @type {BillingService}
+     */
+    const instance = this // eslint-disable-line
+  }
+
+  /**
+   * @param {import('./billing-types').StripeCustomerId} customer
+   * @param {import('./billing-types').StripePaymentMethodId} method
+   * @returns {Promise<void>}
+   */
+  async savePaymentMethod (customer, method) {
+
+  }
+}
+
+/**
+ * @typedef {import("./billing-types").CustomersService} CustomersService
+ */
+
+/**
+ * @typedef {import("./billing-types").Customer} Customer
+ */
+
+/**
+ * @typedef {import('stripe').Stripe['customers']} StripeComCustomers
+ */
+
+/**
+ * @typedef {object} StripeComCustomersForGetOrCreate
+ * @property {StripeComCustomers['create']} create
+ * @property {StripeComCustomers['search']} search
+ */
+
+/**
+ * @typedef {object} StripeComForCustomersService
+ * @property {StripeComCustomersForGetOrCreate} customers
+ */
+
+/**
+ * A CustomersService that uses stripe.com for storage
+ */
+export class StripeCustomersService {
+  /**
+   * @param {StripeComForCustomersService} stripe
+   */
+  static create (stripe) {
+    return new StripeCustomersService(stripe)
+  }
+
+  /**
+   * @param {StripeComForCustomersService} stripe
+   * @protected
+   */
+  constructor (stripe) {
+    /** @type {StripeComForCustomersService} */
+    this.stripe = stripe
+    /**
+     * @type {CustomersService}
+     */
+    const instance = this // eslint-disable-line
+  }
+
+  /**
+   * @param {{id: string}} user
+   * @returns {Promise<Customer>}
+   */
+  async getOrCreateForUser (user) {
+    const userIdMetadataProperty = 'web3StorageUserId'
+    // const searchQuery = `metadata["${userIdMetadataProperty}"]:"${user.id}"`
+    const searchQuery = `metadata["${userIdMetadataProperty}"]:"${user.id}"`
+    // console.log({ searchQuery })
+    const searchResponse = await this.stripe.customers.search({
```

Comment:
Unfortunately the search index doesn't update for like 5+ sec after writes. So this hope to do a version of this without a db change aint gonna work. I think I'll need a db change to store [user.id, stripeCustomer.id]